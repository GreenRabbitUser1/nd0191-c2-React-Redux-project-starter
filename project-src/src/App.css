#root {
    width: 100%;
    height: 100%;
    text-align: center;
    padding-bottom: 3em;
}

.logo {
    height: 6em;
    padding: 1.5em;
    will-change: filter;
    transition: filter 300ms;
}

.logo:hover {
    filter: drop-shadow(0 0 2em #646cffaa);
}

.logo.react:hover {
    filter: drop-shadow(0 0 2em #61dafbaa);
}

@keyframes logo-spin {
    from {
        transform: rotate(0deg);
    }

    to {
        transform: rotate(360deg);
    }
}

@media (prefers-reduced-motion: no-preference) {
    a:nth-of-type(2) .logo {
        animation: logo-spin infinite 20s linear;
    }
}

.card {
    padding: 2em;
}

.read-the-docs {
    color: #888;
}

.login-form-wrap {
    width: 100%;
    display: flex;
    justify-content: center;
}

.login-form {
    display: flex;
    flex-flow: column;
    gap: 1em;
    justify-content: center;
    align-items: center;
    border: 1px solid rgb(15 0 255 / 16%);
    padding: 2em;
    padding-top: 1em;
    border-radius: 20px;
    background: rgb(128 177 240 / 18%);
    box-shadow: 0px 0px 6px 2px rgb(31 94 215 / 18%);
    /* width: max-content; */
}

.login-form>div {
    display: flex;
    flex-flow: column;
    gap: 0.8em;
}

.login-form input {
    border: 1px solid rgba(168, 168, 168, 0.348);
    background: rgb(73, 73, 73);
    border-radius: 4px;
    height: 1.9em;
    /* width: 39ch; */
    color: white;
}

.authError {
    background: #a43c3c;
    color: white;
    padding: 0.5em 1em;
    border-radius: 4px;
}

ul.header-list-left,
ul.header-list-right,
ul.profile-list {
    display: flex;
    flex-flow: row;
    gap: 3em;
    list-style-type: none;
}

.header {
    display: flex;
    flex-flow: row wrap;
    justify-content: space-between;
    margin: 2em;
}

.header a {
    color: #90ff9b !important;
    border-bottom: 2px solid transparent;
    cursor: pointer;
}

.header a:hover {
    border-bottom: 2px solid #90ff9b !important;
}

.active_page {
    border-bottom: 2px solid #90ff9b !important;
}

.profile-list-wrap {
    position: relative;
}

.profile-list {
    gap: 0.4em !important;
    /* display: flex;
    flex-flow: row; */
}

.profile-dropdown-wrap {
    display: flex;
    flex-flow: row;
    align-items: center;
}

.profile-dropdown {
    display: flex;
    align-items: center;
    transition: transform 0.2s;
    cursor: pointer;
}

.profile-dropdown-open {
    transform: rotate(180deg);
}

.logout {
    background: #4d4d4d;
    position: absolute;
    right: 0;
    border-radius: 6px;
    padding: 0.3em 0.9em;
    margin-top: 10px;
    cursor: pointer;
}

.logout:hover {
    background: #6d6d6d !important;
}

.add-poll-form {
    display: flex;
    flex-flow: column;
    gap: 2em;
    justify-content: center;
    align-items: center;
}

.add-poll-form textarea {
    height: 5em;
    width: 40ch;
    border-radius: 6px;
    background: #414141;
    color: white;
    padding: 8px;
}

.form-top {
    display: flex;
    flex-flow: column;
    justify-content: center;
    align-items: center;
    font-size: xx-large;
}

.form-bottom {
    display: flex;
    flex-flow: row wrap;
    gap: 1em;
}

.form-bottom > div {
    display: flex;
    flex-flow: column;
}

.poll-widget-wrap {
    border: 1px solid #3a7a47;
    padding: 1em;
    display: flex;
    flex-flow: column;
    gap: 1em;
    max-width: 310px;
    justify-content: flex-start;
    border-radius: 10px;
    box-shadow: 0 0 6px 2px #182d20;
    background: #8fd18f0a;
    min-width: 300px;
}

.polls {
    display: flex;
    flex-flow: row wrap;
    justify-content: center;
    gap: 3em;
    max-width: 80vw;
}

.home-wrap {
    display: flex;
    flex-flow: column;
    align-items: center;
}

.poll-author label {
    font-size: small;
}

.poll-question {
    font-size: x-large;
}

.poll-body {
    display: flex;
    flex-flow: column;
    gap: 1em;
    height: 100%;
    justify-content: space-between;
}

.poll-options {
    display: flex;
    flex-flow: column;
    gap: 1em;
    justify-content: center;
}

.option-answered {
    max-width: 95%;
    padding: 0.2em 0.5em;
    background: #323232;
    border-radius: 6px;
    position: relative;
    border: 1px solid black;
    cursor: default;
    display: flex;
    flex-flow: column;
    gap: 0.2em;
}

.votes {
    position: absolute;
    width: 25%;
    height: 100%;
    background: #00fffa3b;
    left: 0;
    top: 0;
    border-radius: 6px;
}

/* .poll-options > div {
    max-width: 50%;
} */

.option-outcome {
    font-size: x-small;
}

.poll-category {
    display: flex;
    flex-flow: column;
    justify-content: center;
    gap: 1em;
}

.poll-wrap {
    display: flex;
    flex-flow: row wrap;
    gap: 2em;
}

.poll-category-name {
    font-size: xx-large;
}

.welcome-message {
    min-width: 25%;
    border: 1px solid #56a3b841;
    padding: 10px;
    border-radius: 6px;
    background: #357a917a;
    margin-bottom: 20px;
}

.view-wrap {
    position: relative;
    width: 85%;
    text-align: left;
    display: flex;
    flex-flow: column;
    justify-content: flex-start;
    align-items: center;
}

.view-filter-top {
    position: relative;
    width: 85%;
    text-align: left;
    display: flex;
    flex-flow: row;
    gap: 0.3em;
    justify-content: flex-start;
    align-items: center;
    cursor: pointer;
}

.view-filter-icon {
    transition: transform 0.2s;
}

.view-filter-open {
    transform: rotate(180deg);
}

.view-wrap svg:first-child {
    height: 1.5em !important;
    width: 1.5em !important;
}

.view-wrap span {
    font-size: larger;
}

.view-options-wrap {
    position: relative;
    width: 85%;
    text-align: left;
    display: flex;
    flex-flow: row;
    justify-content: flex-start;
    align-items: center;
}

.view-options {
    position: absolute;
    z-index: 2;
    background: #424242;
    top: 0;
    left: 0;
    padding: 10px;
    border-radius: 6px;
    display: flex;
    flex-flow: column;
    gap: 1em;
}

.view-option > input[type="checkbox"] {
    transform: scale(1.5);
}

.view-option {
    display: flex;
    flex-flow: row;
    gap: 0.3em;
}

.poll-author-wrap {
    display: flex;
    flex-flow: row;
    justify-content: center;
    gap: 0.5em;
    align-items: center;
}

.poll-author-avatar {
    height: 1.7em;
    width: 1.7em;
    border-radius: 50%;
    border: 1px solid white;
}

.user-selected-answer {
    width: 10%;
    position: absolute;
    top: 50%;
    right: 0px;
}

.user-selected-answer svg {
    height: 1.5em;
    width: 1.5em;
}

.option-text-wrap {
    display: flex;
    flex-flow: row;
    justify-content: space-around;
    position: relative;
    width: 100%;
}

.poll-text {
    width: 70%;
}

.leaderboard-table-wrap {
    display: flex;
    flex-flow: row;
    justify-content: center;
}

.leaderboard-table {
    border: 1px solid white;
    border-collapse: collapse;
}

.leaderboard-table th {
    padding: 6px 12px;
    background: #1f5531;
    border: 1px solid white;
}

.leaderboard-table td {
    padding: 6px 12px;
    border: 1px solid white;
    color: #e2e2e2;
}

.leaderboard-user-wrap {
    display: flex;
    flex-flow: row;
    justify-content: flex-start;
    align-items: center;
    gap: 0.5em;
}

.leaderboard-user-wrap img {
    height: 1.5em;
    width: 1.5em;
    border: 1px solid white;
    border-radius: 50%;
}

.leaderboard-current-user {
    background: #ffff9236;
}

.poll-header {
    position: relative;
}

svg.widget-nav:hover {
    fill: #45a468;
}

.widget-nav {
    height: 1.4em;
    width: 1.4em;
    position: absolute;
    top: 0;
    right: 0;
    fill: #85ffb2;
    stroke: #85ffb2;
    transition: 0.15s all;
}

.poll-page-widget-wrap {
    display: flex;
    justify-content: center;
}

.submitting-answer {
    display: flex;
    justify-content: center;
    align-items: center;
    height: 100%;
    flex-flow: column;
}

form.profile-wrap {
    display: flex;
    flex-flow: column;
    gap: 1em;
    align-items: center;
}

.main-info-wrap {
    display: flex;
    flex-flow: column;
    gap: 1em;
    justify-content: center;
    align-items: center;
}

.field-wrap {
    display: flex;
    flex-flow: column;
    align-items: center;
}

.field-wrap input {
    height: 1.8em;
    width: 29ch;
}

.profile-wrap .main-info-wrap img {
    width: 5em;
    height: 5em;
    border: 1px solid pink;
    margin-bottom: 1em;
    border-radius: 50%;
    background: #ffaebc4f;
}

.mt-2 {
    margin-top: 2em;
    width: max-content;
}