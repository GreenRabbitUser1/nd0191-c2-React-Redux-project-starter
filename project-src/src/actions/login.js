import * as API from '../../_DATA';

export const LOGIN_REQUEST = 'LOGIN_REQUEST';
export const LOGIN_FAILURE = 'LOGIN_FAILURE';
export const LOGIN_ERROR = 'LOGIN_ERROR';
export const LOGIN_SUCCESS = 'LOGIN_SUCCESS';
export const SESSION_RESTORE = 'SESSION_RESTORE';
export const ATTEMPT_RESTORE = 'ATTEMPT_RESTORE';

export function setUser(user){
    return {
        type: LOGIN_REQUEST,
        user
    };
}

export function attemptLogin({username, password}){
    return async (dispatch) => {
        dispatch({
            type: LOGIN_REQUEST
        });
        try {

            const {user, getUserError} = await getUser(username, password);
            if (getUserError && getUserError !== null){
                dispatch(getUserError);
                return false;
            }

            //  SUCCESS! Username and password matches, user has been authenticated
            let sessionToken = await fetch('/api/auth/generate-jwt', {
                body: JSON.stringify({username, password}),
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json'
                }
            });
            if (sessionToken.ok){
                sessionToken = await sessionToken.json();
                console.log('sessionToken:', sessionToken);
                localStorage.setItem('udacity-final-project-token', sessionToken);
                const lst = localStorage.getItem('udacity-final-project-token');
                console.info(lst);
                // console.info('lst: ', lst);
            }
            dispatch({
                type: LOGIN_SUCCESS,
                user
            });
            return true;

        } catch(error) {
            dispatch({
                type: LOGIN_FAILURE,
                error: error.message
            })
            return false;

        }
       
    }
}

async function getUser(username, password){
    let user;
    let error;
    const response = await API._getUsers();
    if (!response || response === null){
        error = {
            type: LOGIN_FAILURE,
            error: 'Did not find any users'
        };
    }
    try {
        user = response[username];
    } catch(e){}
    if (!user || user === null){
        error = {
            type: LOGIN_FAILURE,
            error: 'Did not find user matching this username'
        };
    }
    else {
        //  Check if the password matches
        if (user.password !== password){
            error = {
                type: LOGIN_FAILURE,
                error: 'Password did not match'
            };
        }
    }
    return {
        user,
        getUserError: error
    };
}

export function refreshUser(uId){
    return async (dispatch) => {
        console.log('Refreshing User ...');
        const response = await API._getUsers();
        let user;
        if (response && response !== null){
            user = response[uId];
            dispatch({
                type: LOGIN_SUCCESS,
                user
            });
            console.log('Refreshed.');
            return true;
        }
    }
}

export function attemptSessionRestore(){

    return async (dispatch) => {
        //  Check if the user has a logged in cookie
        let validSession = false;
        const sessionToken = localStorage.getItem('udacity-final-project-token');
        if (sessionToken && sessionToken !== null){
            console.log('There is a valid session token', sessionToken);
            let decodedToken = await fetch('/api/auth/decode-jwt', {
                body: JSON.stringify({
                    token: sessionToken
                }),
                method: 'POST',
                headers: {
                    'Content-Type': 'application/json'
                }
            });
            if (decodedToken.ok){
                decodedToken = await decodedToken.json();
                //  With the decodedToken the following are now available: username, password, iat, exp
                //  Get the user, set the state in the store, redirect to / if on /login
                console.log('decodedToken', decodedToken);
                const {user, getUserError} = await getUser(decodedToken.username, decodedToken.password);
                if (getUserError && getUserError !== null){
                    dispatch(getUserError);
                    return validSession;
                }
                dispatch({
                    type: LOGIN_SUCCESS,
                    user
                });
                dispatch({
                    type: ATTEMPT_RESTORE
                });
                validSession = true;
            }
        }

        return validSession;

        // dispatch({
        //     type: SESSION_RESTORE
        // })
    }

}