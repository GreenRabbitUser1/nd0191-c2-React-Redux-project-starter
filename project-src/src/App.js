import { useState, useEffect } from 'react'
import './App.css'
import { Route, Routes, useNavigate } from 'react-router-dom';
import MainLayout from './components/MainLayout';
import HomePage from './pages/HomePage';
import LoginPage from './pages/LoginPage';
import LeaderboardPage from './pages/LeaderboardPage';
import AddPollPage from './pages/AddPollPage';
import ProfilePage from './pages/ProfilePage';
import PollPage from './pages/PollPage';
import { connect } from 'react-redux';
import RequireAuth from './components/RequireAuth';
import { Navigate } from 'react-router-dom';

function App(props) {

    return (
            <Routes>
                {/* Require users to authenticate */}
                <Route path="/login" element={<LoginPage />} /> 

                <Route element={<RequireAuth/>}>
                    {/* Users can answer polls on the Dashboard of the HomePage  */}
                    <Route path="/" element={<MainLayout page={<HomePage />}></MainLayout>} />

                    {/* Users can view the Leaderboard with the number of answered polls and created polls by user */}
                    <Route exact path="/leaderboard" element={<MainLayout page={<LeaderboardPage />}></MainLayout>} />

                    {/* Users can create new polls for other users to answer */}
                    <Route exact path="/add" element={<MainLayout page={<AddPollPage />}></MainLayout>} />

                    {/* Users can create new polls for other users to answer */}
                    <Route path="/question/:id" element={<MainLayout page={<PollPage />}></MainLayout>} />

                    {/* Users can view their profile settings and perform some updates such as password reset */}
                    <Route path="/profile" element={<MainLayout page={<ProfilePage />}></MainLayout>} />
                </Route>

                {/* Catch-all route for 404 - currently navigates back to login screen - fix to only navigate if user is not authenticated */}
                <Route path="*" element={<Navigate to='/login' />} />

            </Routes>
    )

    
}

const mapStateToProps = (state) => ({
    user: state.user
});

export default connect(mapStateToProps)(App);
