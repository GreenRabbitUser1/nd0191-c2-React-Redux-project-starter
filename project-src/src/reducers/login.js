import { LOGIN_REQUEST, LOGIN_SUCCESS, LOGIN_FAILURE } from "../actions/login";
import { LOGOUT_USER } from "../actions/logout";
import { SAVE_USER } from "../actions/users";
import { ATTEMPT_RESTORE } from "../actions/login";

export function user(state = null, action){
    switch(action.type){
        case LOGIN_SUCCESS:
            return action.user;
        case LOGOUT_USER: 
            return null;
        case SAVE_USER: 
            return action.user;
        default: 
            return state;
    }

}

export function authDone(state = false, action){
    switch(action.type){
        case ATTEMPT_RESTORE: 
            return true;
        default:
            return state;
    }
}

export function authError(state = '', action){
    switch(action.type){
        case LOGIN_FAILURE:
            return action.error;
        default: 
            return state;
    }
}