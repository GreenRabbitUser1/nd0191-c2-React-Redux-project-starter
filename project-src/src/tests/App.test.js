import React from "react";
import {render, screen, fireEvent, act, waitFor} from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import App from "../App.js";
import { Provider } from "react-redux";
import { createStore } from "redux";
import rootReducer from '../reducers/index.js';
import { thunk } from 'redux-thunk'
import { applyMiddleware } from "redux";
import { setUser } from "../actions/login.js";
import store from "../store.js";
import Header from "../components/Header.js";


// const store = createStore(rootReducer, applyMiddleware(thunk));


describe("App Load", () => {
    it("should render App", async () => {
        // Render the component

        const component = await act(async () => {
            render(
                <Provider store={store}>
                    <Router>
                        <App />
                    </Router>
                </Provider>
        )});

        expect(screen.getByText('Workplace Polls App')).toBeInTheDocument();

    });
});

describe("Login Page", () => {
    it("should find login fields and add values to them", async () => {
        // Render the component
        
        const component = await act(async () => {
            render(
                <Provider store={store}>
                    <Router>
                        <App />
                    </Router>
                </Provider>
        )});

        // Get the input fields
        const usernameInput = screen.getByLabelText(/username/i);
        const passwordInput = screen.getByLabelText(/password/i);

        // Simulate user typing into the fields
        fireEvent.change(usernameInput, {
            target: {
                value: "sarahedo"
            }
        });
        fireEvent.change(passwordInput, {
            target: {
                value: "password123"
            }
        });

        // Assert the input values
        expect(usernameInput.value).toBe("sarahedo");
        expect(passwordInput.value).toBe("password123");
    });
});

describe("Header", () => {
    it("should render the Header", async () => {
        // Render the component

        const component = render(
            <Provider store={store}>
                <Router>
                    <Header />
                </Router>
            </Provider>
        );

        // Assert the input values
        expect(component).toBeDefined();
        // expect(component).toMatchSnapshot();
    });
});