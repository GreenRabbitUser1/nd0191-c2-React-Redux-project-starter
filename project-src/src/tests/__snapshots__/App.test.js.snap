// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App Render app component 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div />
  </body>,
  "container": <div />,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`AppLoad should render App 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="header"
      >
        <ul
          class="header-list-left"
        >
          <li>
            <a
              class="active_page"
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              class=""
              href="/add"
            >
              Add Poll
            </a>
          </li>
          <li>
            <a
              class=""
              href="/leaderboard"
            >
              Leaderboard
            </a>
          </li>
        </ul>
        <ul
          class="header-list-right"
        >
          <li
            class="profile-list-wrap"
          >
            <ul
              class="profile-list"
            >
              <li>
                <a
                  href="/profile"
                />
              </li>
              <li
                class="profile-dropdown-wrap"
              >
                <svg
                  class="
                                    profile-dropdown
                                    
                                "
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                  />
                </svg>
              </li>
            </ul>
          </li>
        </ul>
      </div>
      <main>
        <div
          class="home-wrap"
        >
          <div
            class="view-wrap"
          >
            <div
              class="view-filter-top"
            >
              <svg
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 512 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M256 105c-101.8 0-188.4 62.4-224 151 35.6 88.6 122.2 151 224 151s188.4-62.4 224-151c-35.6-88.6-122.2-151-224-151zm0 251.7c-56 0-101.8-45.3-101.8-100.7S200 155.3 256 155.3 357.8 200.6 357.8 256 312 356.7 256 356.7zm0-161.1c-33.6 0-61.1 27.2-61.1 60.4s27.5 60.4 61.1 60.4 61.1-27.2 61.1-60.4-27.5-60.4-61.1-60.4z"
                />
              </svg>
              <span>
                Filter View
              </span>
              <svg
                class="
                        view-filter-icon 
                        
                    "
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                />
              </svg>
            </div>
          </div>
          <div
            class="polls"
          >
            <div
              class="poll-category"
            >
              <div
                class="poll-category-name"
              >
                Unanswered Polls
              </div>
              <div
                class="poll-wrap"
              />
            </div>
            <div
              class="poll-category"
            >
              <div
                class="poll-category-name"
              >
                Answered Polls
              </div>
              <div
                class="poll-wrap"
              />
            </div>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <div
      class="header"
    >
      <ul
        class="header-list-left"
      >
        <li>
          <a
            class="active_page"
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            class=""
            href="/add"
          >
            Add Poll
          </a>
        </li>
        <li>
          <a
            class=""
            href="/leaderboard"
          >
            Leaderboard
          </a>
        </li>
      </ul>
      <ul
        class="header-list-right"
      >
        <li
          class="profile-list-wrap"
        >
          <ul
            class="profile-list"
          >
            <li>
              <a
                href="/profile"
              />
            </li>
            <li
              class="profile-dropdown-wrap"
            >
              <svg
                class="
                                    profile-dropdown
                                    
                                "
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                />
              </svg>
            </li>
          </ul>
        </li>
      </ul>
    </div>
    <main>
      <div
        class="home-wrap"
      >
        <div
          class="view-wrap"
        >
          <div
            class="view-filter-top"
          >
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 512 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M256 105c-101.8 0-188.4 62.4-224 151 35.6 88.6 122.2 151 224 151s188.4-62.4 224-151c-35.6-88.6-122.2-151-224-151zm0 251.7c-56 0-101.8-45.3-101.8-100.7S200 155.3 256 155.3 357.8 200.6 357.8 256 312 356.7 256 356.7zm0-161.1c-33.6 0-61.1 27.2-61.1 60.4s27.5 60.4 61.1 60.4 61.1-27.2 61.1-60.4-27.5-60.4-61.1-60.4z"
              />
            </svg>
            <span>
              Filter View
            </span>
            <svg
              class="
                        view-filter-icon 
                        
                    "
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 448 512"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
              />
            </svg>
          </div>
        </div>
        <div
          class="polls"
        >
          <div
            class="poll-category"
          >
            <div
              class="poll-category-name"
            >
              Unanswered Polls
            </div>
            <div
              class="poll-wrap"
            />
          </div>
          <div
            class="poll-category"
          >
            <div
              class="poll-category-name"
            >
              Answered Polls
            </div>
            <div
              class="poll-wrap"
            />
          </div>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`Header should render the Header 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body
    style="justify-content: center;"
  >
    <div>
      <div
        class="header"
      >
        <ul
          class="header-list-left"
        >
          <li>
            <a
              class=""
              href="/"
            >
              Home
            </a>
          </li>
          <li>
            <a
              class=""
              href="/add"
            >
              Add Poll
            </a>
          </li>
          <li>
            <a
              class=""
              href="/leaderboard"
            >
              Leaderboard
            </a>
          </li>
        </ul>
        <ul
          class="header-list-right"
        >
          <li
            class="profile-list-wrap"
          >
            <ul
              class="profile-list"
            >
              <li>
                <a
                  data-testid="header-user-name"
                  href="/profile"
                />
              </li>
              <li
                class="profile-dropdown-wrap"
              >
                <svg
                  class="
                                    profile-dropdown
                                    
                                "
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 448 512"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                  />
                </svg>
              </li>
            </ul>
          </li>
        </ul>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="header"
    >
      <ul
        class="header-list-left"
      >
        <li>
          <a
            class=""
            href="/"
          >
            Home
          </a>
        </li>
        <li>
          <a
            class=""
            href="/add"
          >
            Add Poll
          </a>
        </li>
        <li>
          <a
            class=""
            href="/leaderboard"
          >
            Leaderboard
          </a>
        </li>
      </ul>
      <ul
        class="header-list-right"
      >
        <li
          class="profile-list-wrap"
        >
          <ul
            class="profile-list"
          >
            <li>
              <a
                data-testid="header-user-name"
                href="/profile"
              />
            </li>
            <li
              class="profile-dropdown-wrap"
            >
              <svg
                class="
                                    profile-dropdown
                                    
                                "
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 448 512"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M207.029 381.476L12.686 187.132c-9.373-9.373-9.373-24.569 0-33.941l22.667-22.667c9.357-9.357 24.522-9.375 33.901-.04L224 284.505l154.745-154.021c9.379-9.335 24.544-9.317 33.901.04l22.667 22.667c9.373 9.373 9.373 24.569 0 33.941L240.971 381.476c-9.373 9.372-24.569 9.372-33.942 0z"
                />
              </svg>
            </li>
          </ul>
        </li>
      </ul>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
